From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 16 Jun 2014 02:03:10 +0100
Subject: PCI: msi_desc: remove unused kobject
Bug-Debian: https://bugs.debian.org/748595

This is unused since commit 1c51b50c2995 ('PCI/MSI: Export MSI mode using
attributes, not kobjects').

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
For Debian, this also avoids an ABI change.
---
--- a/include/linux/msi.h
+++ b/include/linux/msi.h
@@ -45,8 +45,6 @@ struct msi_desc {
 
 	/* Last set MSI message */
 	struct msi_msg msg;
-
-	struct kobject kobj;
 };
 
 /*
--- a/drivers/pci/msi.c
+++ b/drivers/pci/msi.c
@@ -327,17 +327,6 @@ static void free_msi_irqs(struct pci_dev
 				iounmap(entry->mask_base);
 		}
 
-		/*
-		 * Its possible that we get into this path
-		 * When populate_msi_sysfs fails, which means the entries
-		 * were not registered with sysfs.  In that case don't
-		 * unregister them.
-		 */
-		if (entry->kobj.parent) {
-			kobject_del(&entry->kobj);
-			kobject_put(&entry->kobj);
-		}
-
 		list_del(&entry->list);
 		kfree(entry);
 	}
